cmd/juju/config/config.go

```diff
@@ -0,0 +1,199 @@
+
+	// Flag values
+	ConfigFile cmd.FileVar
+	reset      []string // Holds the keys to be reset until parsed.
+
+	// Fields to be set during initialisation
+	Actions     []ConfigAction // The action which we want to handle, set in Init.
+	KeyToGet    string
+	KeysToReset []string
+	ValsToSet   map[string]string
+}
+
+// Info - to be implemented by child command
+
+// SetFlags implements cmd.Command.SetFlags.
+func (c *ConfigCommandBase) SetFlags(f *gnuflag.FlagSet) {
+	// Set the --format and -o flags
+	c.Out.AddFlags(f, "yaml", map[string]cmd.Formatter{
+		"json": cmd.FormatJson,
+		"yaml": cmd.FormatYaml,
+		// "tabular":     formatConfigTabular,
+		// "constraints": formatConstraints,
+	})
+
+	f.Var(&c.ConfigFile, "file", "path to yaml-formatted configuration file")
+	// f.BoolVar(&c.IgnoreReadOnlyFields, "ignore-read-only-fields", false, "Ignore read only fields that might cause errors to be emitted while processing yaml documents")
+	f.Var(cmd.NewAppendStringsValue(&c.reset), "reset", "Reset the provided comma delimited keys, deletes keys not in the model config")
+}
+
+// Init provides a basic implementation of cmd.Command.Init.
+// This only parses the actual key/value arguments in the command. Some child
+// commands will also need to specify what is being configured (e.g. the app
+// name in the case of `juju config`). In this case, the child should define
+// its own Init command, where it strips the required arguments and passes the
+// rest to the parent.
+func (c *ConfigCommandBase) Init(args []string) error {
+	// Check if --file has been specified
+	if c.ConfigFile.Path != "" {
+		c.Actions = append(c.Actions, SetFile)
+	}
+
+	// Check if --reset has been specified
+	if len(c.reset) != 0 {
+		c.Actions = append(c.Actions, Reset)
+		err := c.parseResetKeys()
+		if err != nil {
+			return errors.Trace(err)
+		}
+	}
+
+	// The remaining arguments are divided into setKeys (args containing `=`)
+	// and getKeys, so we can work out what the user is trying to do.
+	var setKeys, getKeys []string
+	for _, arg := range args {
+		splitArg := strings.Split(arg, "=")
+		if len(splitArg) > 1 {
+			setKeys = append(setKeys, splitArg[0])
+		} else {
+			getKeys = append(getKeys, arg)
+		}
+	}
+
+	if len(setKeys) == 0 {
+		switch len(getKeys) {
+		case 0:
+			if len(c.Actions) == 0 {
+				// Nothing specified - get full config
+				c.Actions = append(c.Actions, GetAll)
+			}
+		case 1:
+			// Get the single key specified
+			c.Actions = append(c.Actions, GetOne)
+			c.KeyToGet = args[0]
+		default:
+			// Trying to get multiple keys - error
+			return errors.New("cannot get multiple keys at once")
+		}
+	} else {
+		if len(getKeys) == 0 {
+			// Set specified keys to given values
+			c.Actions = append(c.Actions, Set)
+			settings, err := keyvalues.Parse(args, true)
+			if err != nil {
+				return err
+			}
+			c.ValsToSet = settings
+		} else {
+			// Looks like user is trying to get & set simultaneously
```

Comment:
We should try to avoid nest conditions to simplify process.
So I would return invalid case first,
then other cases:
```go
if len(getKeys) > 0 && len(setKeys) > 0 {return error}
if len(getKeys) == 0 && len(setKeys) == 0 {GetAll}
if len(getKeys) > 0 {Get}
if len(SetKeys) > 0 {Set}
return nil
```